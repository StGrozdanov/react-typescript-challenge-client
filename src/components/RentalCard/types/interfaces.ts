import { Rental } from "../../../types/interfaces"

export interface RentalDataProps {
    data: Rental
}

export interface RentalDescriptionProps {
    rentalName: string
}

export interface RentalImageProps {
    imgURL: string,
    altImgURL: string,
    className?: string,
}