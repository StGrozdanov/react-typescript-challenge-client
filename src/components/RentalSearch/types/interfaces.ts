import { Rental } from "../../../types/interfaces";

export interface RentalSearchProps {
  keyword: string
}

export interface RentalSearchResultsProps {
  data: Rental[],
  keyword: string,
}

export interface RentalSearchLocationProps {
  location?: string,
  search?: string,
}