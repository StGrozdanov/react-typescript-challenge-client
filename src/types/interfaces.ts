export interface Rental {
    id: number,
    name: string,
    imgURL: string,
}
