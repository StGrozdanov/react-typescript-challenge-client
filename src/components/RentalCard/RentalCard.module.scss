.rentalCard {
    display: flex;
    padding: 2rem 0.5rem;
    justify-content: flex-start;
    align-items: center;
}