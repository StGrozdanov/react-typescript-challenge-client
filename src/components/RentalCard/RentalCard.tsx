import styles from './RentalCard.module.scss'
import RentalCardImage from './modules/RentalCardImage/RentalCardImage'
import RentalCardDescription from './modules/RentalCardDescription/RentalCardDescription'
import { RentalDataProps } from './types/interfaces'

const RentalCard: React.FC<RentalDataProps> = ({ data }: RentalDataProps) => {
    return (
        <div data-testid="rentalCard" className={styles.rentalCard}>
            <RentalCardImage
                imgURL={data.imgURL}
                altImgURL="https://getlogovector.com/wp-content/uploads/2020/10/outdoorsy-inc-logo-vector.png"
            />
            <RentalCardDescription rentalName={data.name} />
        </div>
    )
}

export default RentalCard;