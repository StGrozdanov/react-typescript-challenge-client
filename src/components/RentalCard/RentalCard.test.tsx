import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'; // Import Jest DOM extensions
import RentalCard from './RentalCard';
import { Rental } from '../../types/interfaces';

describe('RentalCard Component', () => {
    const mockedRental: Rental = {
        id: 342669,
        name: 'Vintage Camper Van 1985 GMC Vandurah',
        imgURL: 'https://res.cloudinary.com/outdoorsy/image/upload/v1644599588/p/rentals/226946/images/exzlo5wgf2yiwfjjo7zi.jpg'
    }

    beforeEach(() => {
        render(<RentalCard data={mockedRental} />);
    });

    test('Renders rental card with image and description', () => {
        const rentalCard = screen.getByTestId('rentalCard');
        const rentalImage = screen.getByTestId('rentalCardImage');
        const rentalDescription = screen.getByTestId('rentalCardDescription');

        // Check if the rental card, image, and description elements are present in the rendered component.
        expect(rentalCard).toBeInTheDocument();
        expect(rentalImage).toBeInTheDocument();
        expect(rentalDescription).toBeInTheDocument();

        // Check if the image has the correct src attribute.
        expect(rentalImage).toHaveAttribute('src', mockedRental.imgURL);

        // Check if the description displays the correct rental name.
        expect(rentalDescription).toHaveTextContent(mockedRental.name);
    });

    test('Matches snapshot', () => {
        const { container } = render(<RentalCard data={mockedRental} />);
        expect(container).toMatchSnapshot();
    });
});