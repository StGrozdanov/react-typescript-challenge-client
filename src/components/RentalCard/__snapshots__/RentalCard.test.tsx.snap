// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RentalCard Component Matches snapshot 1`] = `
<div>
  <div
    class="rentalCard"
    data-testid="rentalCard"
  >
    <div
      class="imageContainer"
    >
      <img
        class="image"
        data-testid="rentalCardImage"
        src="https://res.cloudinary.com/outdoorsy/image/upload/v1644599588/p/rentals/226946/images/exzlo5wgf2yiwfjjo7zi.jpg"
      />
    </div>
    <p
      class="description"
      data-testid="rentalCardDescription"
    >
      Vintage Camper Van 1985 GMC Vandurah
    </p>
  </div>
</div>
`;
